packages/perseus/src/widgets/interactive-graphs/locked-line.tsx

```diff
@@ -0,0 +1,66 @@
+import {color as wbColor, spacing} from "@khanacademy/wonder-blocks-tokens";
+import {Point, Line} from "mafs";
+import * as React from "react";
+
+import type {LockedLineType} from "../../perseus-types";
+
+const LockedLine = (props: LockedLineType) => {
+    const {color, lineStyle, kind, points, showStartPoint, showEndPoint} =
+        props;
+    const [point1, point2] = points;
+
+    let line;
+
+    if (kind === "line" || kind === "segment") {
+        const LineType = kind === "segment" ? Line.Segment : Line.ThroughPoints;
+        line = (
+            <LineType
+                point1={point1.coord}
+                point2={point2.coord}
+                color={wbColor[color]}
+                style={lineStyle}
+            />
+        );
+    } else {
+        // TODO(LEMS-1928): Implement rays
+        return null;
+    }
+
+    return (
+        <g className="locked-line">
+            {line}
+            {showStartPoint && (
+                <Point
+                    x={point1.coord[0]}
+                    y={point1.coord[1]}
+                    svgCircleProps={{
+                        style: {
+                            fill: point1.filled
+                                ? wbColor[point1.color]
+                                : wbColor.white,
+                            stroke: wbColor[point1.color],
+                            strokeWidth: spacing.xxxxSmall_2,
+                        },
+                    }}
+                />
+            )}
+            {showEndPoint && (
+                <Point
+                    x={point2.coord[0]}
+                    y={point2.coord[1]}
+                    svgCircleProps={{
+                        style: {
+                            fill: point2.filled
+                                ? wbColor[point2.color]
```

Comment:
I don't think there's a way for me to test the visual style of this circle.

Reply:
I was thinking you might be able to add a test that uses a snapshot... but I'm not sure how busy that'd be. 🤔   Someone else might have an idea.